ystia/yorc
# Feature/gh 248 bootstrap config save

# Pull Request description

1. Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
2. Please read our contribution guidelines
3. PR with a decent set of unit tests have more chance to be reviewed and merged quickly

## Description of the change
Interactive config done by yorc bootstrap command is now automatically exported to a file having the deployment name/ID, into the inputsaves/ directory. deployment id can now be specified using the -n flag on the yorc bootstrap command. 
Interactive config can now be loaded using yorc bootstrap -v option to specify a previously exported file. 
A new option has been added, allowing to only export an interactive config and terminate the program without doing the actual bootstrapping. 

While deploying on a Hostspool, the interactive cli will now first ask for the public_address label, and then for others if needed, instead of waiting for the user to enter the public_address key and label by himself.

### What I did
Implemented things said in the previous section.

### How I did it
Slightly changed the code, especially on the inputs.go/initializeInputs function, where i added at the end the code instructions for exporting the interactive config.

### How to verify it
1) do a bootstrap (yorc bootstrap). 
Check that a file has been created with the default deployment ID (based on time) has been created and is valid, into a inputsaves/ dir, created as well.

2) do a bootstrap using the -n flag to specify a name. 
check that a new file has been created into inputsaves/ dir, with the name specified with the -n flag, and is valid as well.

3) do a bootstrap using the -n flag to specify the same name as in step 2.
check that a new file has been created and is valid, with a name like "[specifiedName]1.yaml". 

4) do a bootstrap, loading config using the -v option and one of the files created during steps 1 2 or 3.
check that the bootstrapping is OK.

5) do a bootstrap while setting config_only flag (yorc bootstrap --config_only), check that a file is created and as in step 1, and that the bootstrapping terminate up straight after the export of the interactive config, outputing a message.


### Description for the changelog
Interactive config done by yorc bootstrap command is now automatically exported to a file having the deployment name/ID, into the inputsaves/ directory. deployment id can now be specified using the -n flag on the yorc bootstrap command. 
Interactive config can now be loaded using yorc bootstrap -v option to specify a previously exported file. 
A new option has been added, allowing to only export an interactive config and terminate the program without doing the actual bootstrapping. 

While deploying on a Hostspool, the interactive cli will now first ask for the public_address label, and then for others if needed, instead of waiting for the user to enter the public_address key and label by himself.
## Applicable Issues
#248
